sky/provision/kubernetes/utils.py

```diff
@@ -2395,3 +2396,23 @@ def process_skypilot_pods(
         num_pods = len(cluster.pods)
         cluster.resources_str = f'{num_pods}x {cluster.resources}'
     return list(clusters.values()), jobs_controllers, serve_controllers
+
+
+def get_gpu_resource_key():
+    """Get the GPU resource name to use in kubernetes.
+    The function first checks for an environment variable.
+    If defined, it uses its value; otherwise, it returns the default value.
+    Args:
+        name (str): Default GPU resource name, default is "nvidia.com/gpu".
+    Returns:
+        str: The selected GPU resource name.
+    """
+    # Retrieve GPU resource name from environment variable, if set.
+    # E.g., can be nvidia.com/gpu-h100, amd.com/gpu etc.
+    custom_name = os.getenv('CUSTOM_GPU_RESOURCE_KEY')
+
+    # If the environment variable is not defined, return the default name
+    if custom_name is None:
+        return GPU_RESOURCE_KEY
+
+    return custom_name
```

Comment:
Can simplify logic to:
```suggestion
    # Retrieve GPU resource name from environment variable, if set. Else use default.
    # E.g., can be nvidia.com/gpu-h100, amd.com/gpu etc.
    return os.getenv('CUSTOM_GPU_RESOURCE_KEY', default = GPU_RESOURCE_KEY)
```